build.gradle

```diff
@@ -47,131 +54,85 @@ buildscript {
+
     def deployVersion = properties['DeployVersion']
     if (deployVersion != null) version = deployVersion
     if (useKotlinSnapshot) {
         kotlin_version = '1.2-SNAPSHOT'
     }
 
-    def name = it.name
+    def projectName = it.name
     repositories {
         /*
          * google should be first in the repository list because some of the play services
          * transitive dependencies was removed from jcenter, thus breaking gradle dependency resolution
          */
-        if (name == "kotlinx-coroutines-play-services") {
+        if (projectName == "kotlinx-coroutines-play-services") {
             google()
         }
         jcenter()
         maven { url "https://kotlin.bintray.com/kotlin-dev" }
         maven { url "https://kotlin.bintray.com/kotlin-eap" }
         maven { url "https://kotlin.bintray.com/kotlinx" }
     }
-}
-
-
-// Report Kotlin compiler version when building project
-println("Using Kotlin compiler version: $org.jetbrains.kotlin.config.KotlinCompilerVersion.VERSION")
-
-// --------------- Configure sub-projects with Kotlin sources ---------------
-
-def sourceless = ['site']
 
-static def platformOf(project) {
-    if (project.name.endsWith("-common")) return "common"
-    if (project.name.endsWith("-js")) return "js"
-    if (project.name.endsWith("-native")) return "native"
-    return "jvm"
-}
+    if (projectName == "kotlinx.coroutines" || projectName == coreModule) return
 
-static def platformLib(base, platform) {
-    if (platform == "jvm") return base
-    return "$base-$platform"
-}
-
-subprojects {
-    if (useKotlinSnapshot) {
-        repositories {
-            maven { url "https://oss.sonatype.org/content/repositories/snapshots" }
-        }
-    }
-    configurations.all {
-        resolutionStrategy {
-            eachDependency { DependencyResolveDetails details ->
-                if (details.requested.group == 'org.jetbrains.kotlin') {
-                    details.useVersion kotlin_version
-                }
-            }
-        }
-    }
-}
+    // Add dependency to core source sets. Core is configured in kx-core/build.gradle
+    evaluationDependsOn(":$coreModule")
+    if (sourceless.contains(projectName)) return
 
-configure(subprojects.findAll { !sourceless.contains(it.name)  }) {
     def platform = platformOf(it)
     apply from: rootProject.file("gradle/compile-${platform}.gradle")
-}
-
-// --------------- Configure sub-projects that are part of the library ---------------
+    
+    dependencies {
+        compile project(":$coreModule")
+        // the only way IDEA can resolve test classes
+        testCompile project(":$coreModule").kotlin.targets.jvm.compilations.test.output.allOutputs
+    }
 
-def internal = sourceless + ['benchmarks', 'knit', 'js-stub', 'stdlib-stubs', 'binary-compatibility-validator']
+    tasks.withType(org.jetbrains.kotlin.gradle.tasks.AbstractKotlinCompile).all {
+        kotlinOptions.freeCompilerArgs += ["-Xuse-experimental=kotlin.Experimental",
```

Comment:
Why it's not using `experimentalAnnotations` list from `experimental.gradle` here?